lib/order.rb

```diff
@@ -0,0 +1,80 @@
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize (id, products, customer, fulfillment_status = :pending )
+    # REMEMBER: to set default value do ()fulfillment_status = :pending) in the params of the initialize method
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = validate_fulfillment_status(fulfillment_status)
+  end
+
+  def validate_fulfillment_status(status)
+    until %i[pending paid processing shipped complete].include?(status)
+      raise ArgumentError.new("invalid fullfillment status, #{status}")
+    end
+    return status
+  end
+
+  def total
+    return 0 if @products.empty?
```

Comment:
Great idea and great way to write this!

Something to think about 🤔 :
I think that this `total` method still runs correctly if you delete this line 21 ;) You're checking the case if `@products` is empty but...

In the rest of the method, if you `sum` over `@products`.... `products_total` will be `0`. And if you multiply `0` to calculate `final_total`... `final_total` continues to (correctly) be `0`. :) It might be interesting to delete this line since the math still works.